gnowledge/gstudio
# Course Unit Updates (Creating page/Uploading File)

Adding resources to Course Unit:
	Now a user can create new page or upload a file while adding resources. These will be added to the collection_set of unit node on save.

Test Cases:

Case 1:
	New Unit:
		1. Create new Page
		2. Upload File/ select from drawer to add resources

Case 2:
	New Unit:
		1. Upload File
		2. Create new Page/ select from drawer to add resources

Case 3:
	New Unit:
		1. Select from drawer to add resources
		2. Create new Page/ Upload file

Case 4:
	Existing unit:
		1. Upload file/Create new Page/ select from drawer to add resources

For Case 1, 2, 3, check whether the url is getting updated after step 1.

UI Updates:
* The count of resources that are in the respective node's collection set is displayed based on its type.
Like, Page(5) File(7) etc
* On click of 'Add/Edit Course Structure', Course Authoring page will be opened in same window/tab. (Earlier, it used to open in new tab)
* Cancel button while creating new page or uploading a file


Validations:
* Check on whether page name is entered while creating new page
* Check on whether a file is selected while uploading file
* Check on whether unit name is entered if its for the first time
* Earlier, had check whether right drawer has elements before saving. This is removed now, as it was restricting user to remove resources from unit.

Known issues:
* If a file exists with same name as uploading file name, file is uploaded but does not gets added to node's collection set
* If same file exists as uploading file, file is not uploaded and eventually does not gets added to node's collection set

P.S: Check workflow of 'topics', as it uses the same ajax calls which have been modified now.
